src/util/src/WaveformUtil.cc

```diff
@@ -47,13 +47,14 @@ int GetThresholdCrossingBeforePeak(const std::vector<double>& waveform, int peak
   int back_window = (lb > 0) ? lb : 0;
 
   if (back_window >= waveform.size()) {
-    Log::Die("WaveformUtil: Start of lookback window must be before end of waveform.");
+    warn << "WaveformUtil::GetThresholdCrossingBeforePeak: Start of lookback window not before end of waveform."
+         << newline;
   } else if (back_window >= peakSample) {
-    Log::Die("WaveformUtil: Start of lookback window must be before peak.");
+    warn << "WaveformUtil::GetThresholdCrossingBeforePeak: Start of lookback window not before peak." << newline;
```

Comment:
This message spams the log because you're running this function for all waveforms, not just for ones with pulses that cross threshold.

Reply:
Resolved in commit [7f2f667](https://github.com/rat-pac/ratpac-two/pull/174/commits/7f2f667a30b94310605ef3354052510a93f572c8).

Despite what the diff here says, this debug msg is actually new (it didn't use to die). So one expects to still see _some_ instances of this message despite the fix, but certainly much less spammy.